synapse/storage/databases/main/events_worker.py

```diff
@@ -1854,7 +1854,7 @@ def is_event_next_to_gap_txn(txn: LoggingTransaction) -> bool:
             forward_edge_query = """
                 SELECT 1 FROM event_edges
                 /* Check to make sure the event referencing our event in question is not rejected */
-                LEFT JOIN rejections ON event_edges.event_id == rejections.event_id
+                LEFT JOIN rejections ON event_edges.event_id = rejections.event_id
```

Comment:
`==` is a valid comparison operator in SQLite but not in Postgres.

 - https://sqlite.org/lang_expr.html
 - https://www.postgresql.org/docs/8.2/functions-comparison.html